packages/golden_toolkit/test/multi_screen_golden_test.dart

```diff
@@ -131,6 +131,78 @@ Future<void> main() async {
         expect(tester.binding.window.textScaleFactor, equals(initialTextScaleFactor));
         expect(tester.binding.window.padding, equals(initialViewInsets));
       });
+
+      testGoldens('Should expand scrollable if autoHeight is true', (tester) async {
+        await tester.pumpWidgetBuilder(ListView.builder(
+          itemBuilder: (BuildContext context, int index) {
+            return Container(
+              color: Colors.black.withRed(255 - index * 5),
+              height: 50,
+            );
+          },
+          itemCount: 40,
+        ));
+
+        await multiScreenGolden(
+          tester,
+          'auto_height_list_view',
+          autoHeight: true,
+          devices: [
+            const Device(
+              name: 'anything',
+              size: Size(100, 200),
+              brightness: Brightness.light,
+            )
+          ],
+          skip: !Platform.isMacOS,
+        );
+      });
+
+      testGoldens('Should expand scrollable only if not infinite', (tester) async {
+        await tester.pumpWidgetBuilder(ListView.builder(
+          itemBuilder: (BuildContext context, int index) {
+            return Container(
+              color: Colors.black.withRed(255 - index * 5),
+              height: 50,
+            );
+          },
+        ));
+
+        await multiScreenGolden(
+          tester,
+          'auto_height_infinite_list_view',
+          autoHeight: true,
```

Comment:
I wonder if instead of silently succeeding/clamping, if we should actually throw an assertion failure that autoHeight cannot be used with infinitely sized content at this time.

I think we could iterate in the future on exposing a constraints based configuration for those cases so it could scale up to a supplied constraint.